packages/superflare/src/context.ts

```diff
@@ -0,0 +1,132 @@
+import { AsyncLocalStorage } from "node:async_hooks";
+import type { SuperflareUserConfig } from "./config";
+import { sanitizeModuleName } from "./string";
+
+interface AppContextContainer {
+  ctx?: AppContext;
+}
+
+/**
+ * AppContext represents the context of the current request or job. It is added to the AsyncLocalStorage
+ * so that it can be accessed from anywhere in the current request closure, which means it will not be
+ * leaked between requests.
+ */
+export interface AppContext {
+  appKey?: SuperflareUserConfig["appKey"];
+  env?: any;
+  ctx?: ExecutionContext;
+  database?: {
+    connections: SuperflareUserConfig["database"];
+  };
+  storage?: {
+    disks: SuperflareUserConfig["storage"];
+  };
+  queues?: SuperflareUserConfig["queues"];
+  models?: {
+    [name: string]: any;
+  };
+  jobs?: {
+    [name: string]: any;
+  };
+  events?: {
+    [name: string]: any;
+  };
+  listeners?: Map<string, any[]>;
+  channels?: SuperflareUserConfig["channels"];
+}
+
+export const asyncLocalStorage = new AsyncLocalStorage<AppContextContainer>();
+
+export function getContext(): AppContext {
+  const context = asyncLocalStorage.getStore();
+
+  if (!context || !context.ctx) {
+    throw new Error(
+      "No context found. You must be inside the request lifecycle to access Superflare context."
+    );
+  }
+
+  return context.ctx;
+}
+
+export async function runWithContext<T>(
+  context: AppContext,
+  fn: () => Promise<T>
+) {
+  return await asyncLocalStorage.run({}, async () => {
+    /**
+     * I'm being extra cautious to not set the object value until we're _inside_ the asyncLocalStorage.run
+     * closure. This is to prevent accidental leaks between requests. I don't know if it's necessary.
```

Comment:
TODO: Verify whether this is necessary or if I can just populate an object outside of the run.